### PR TITLE
Fix create child note

### PR DESCRIPTION
**Pull request by TatianaFomina**
- Update editor once page route changes to '/note/:noteId/new'
- Replace `alert` with `console.error` in global errors handler
- Move more logic to `useEditor`

### CODE REVIEW

File: src/application/services/useEditor.ts [hunk 1]
```diff
@@ -99,8 +155,17 @@ export function useEditor({ id, content, isReadOnly, onChange }: UseEditorParams
     }
   }
 
+  /**
+   *
+   * @param data
```
Comment (neSpecc) [inline src/application/services/useEditor.ts:160]: docs missed

File: src/application/services/useNote.ts [hunk 2]
```diff
@@ -139,7 +139,7 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
    * @param content - Note content (Editor.js data)
    * @param parentId - Id of the parent note. If null, then it's a root note
```
Comment (neSpecc) [inline src/application/services/useNote.ts:140]: describe when it can be "undefined"

File: src/presentation/pages/Note.vue [hunk 3]
```diff
@@ -40,7 +41,7 @@ const props = defineProps<{
   /**
    * Parent note id, null for root note
    */
-  parentId: string | null;
+  parentId?: string | null;
```
Comment (neSpecc) [inline src/presentation/pages/Note.vue:44]: describe null/undefined difference meaning

File: src/presentation/pages/Note.vue [hunk 4]
```diff
@@ -77,22 +78,28 @@ function createChildNote(): void {
...
+    /** If new child note is created, refresh editor with empty data */
+    if (props.id === null) {
+      editor.value?.refresh();
+
       useHead({
-        title: noteTitle.value,
+        title: t('note.new'),
+      });
+    } else {
+      watchEffect(() => {
```
Comment (neSpecc) [inline src/presentation/pages/Note.vue:92]: watch inside watch?

File: src/application/services/useNote.ts (hunk 2 above)
Comment (TatianaFomina) [reply src/application/services/useNote.ts:141]: done

File: src/application/services/useEditor.ts (hunk 1 above)
Comment (TatianaFomina) [reply src/application/services/useEditor.ts]: fixed

File: src/presentation/pages/Note.vue (hunk 3 above)
Comment (TatianaFomina) [reply src/presentation/pages/Note.vue]: null is not needed, removed

File: src/presentation/pages/Note.vue (hunk 4 above)
Comment (TatianaFomina) [reply src/presentation/pages/Note.vue]: removed

File: src/application/services/useNote.ts [hunk 5]
```diff
@@ -49,7 +49,7 @@ interface UseNoteComposableState {
   /**
    * Creates/updates the note
    */
-  save: (content: NoteContent, parentId: NoteId | null) => Promise<void>;
+  save: (content: NoteContent, parentId: NoteId | null | undefined) => Promise<void>;
```
[the remark this answers is not in the document]
Comment (TatianaFomina) [reply src/application/services/useNote.ts]: null was incorrect, only `undefined` stays for cases when note does not have parent note

File: src/domain/note.service.ts [hunk 6]
```diff
@@ -47,7 +47,7 @@ export default class NoteService {
    * @param content - Note content (Editor.js data)
    * @param parentId - Id of the parent note. If null, then it's a root note
```
Comment (neSpecc) [inline src/domain/note.service.ts:48]:
```suggestion
   * @param parentId - Id of the parent note. If omitted, then it's a root note
```

**Review by neSpecc — APPROVED** [review]

**Pull request merged**